hub/util/cache_chain.py

```diff
@@ -23,15 +25,48 @@ def get_cache_chain(provider_list: List[StorageProvider], size_list: List[int]):
         ProviderListEmptyError: If the provider list is empty.
         ProviderSizeListMismatch: If the len(size_list) + 1 != len(provider_list)
     """
-    if not provider_list:
+    if not storage_list:
         raise ProviderListEmptyError
-    if len(provider_list) <= 1:
-        return provider_list[0]
-    if len(size_list) + 1 != len(provider_list):
+    if len(storage_list) <= 1:
+        return storage_list[0]
+    if len(size_list) + 1 != len(storage_list):
         raise ProviderSizeListMismatch
-    provider_list.reverse()
+    storage_list.reverse()
     size_list.reverse()
-    store = provider_list[0]
-    for size, cache in zip(size_list, provider_list[1:]):
+    store = storage_list[0]
+    for size, cache in zip(size_list, storage_list[1:]):
         store = LRUCache(cache, store, size)
     return store
+
+
+def generate_chain(
+    base_storage: StorageProvider,
+    memory_cache_size: int,
+    local_cache_size: int,
+    path: str,
+):
+    """Internal function to be used by Dataset, to generate a cache_chain using a base_storage and sizes of memory and local caches.
+
+    Args:
+        base_storage (StorageProvider): The underlying actual storage of the Dataset.
+        memory_cache_size (int): The size of the memory cache to be used in MB.
+        local_cache_size (int): The size of the local filesystem cache to be used in MB.
+        path (str): The location of the dataset.
+
+    Returns:
+        StorageProvider: Returns a cache containing the base_storage along with memory and local cache if a positive size has been specified for them.
+    """
+    dataset_id = str(uuid1())
+    if path:
+        dataset_name = path.split("/")[-1]
+        dataset_id = f"{dataset_name}_{dataset_id}"
+    storage_list: List[StorageProvider] = []
+    size_list: List[int] = []
+    if memory_cache_size > 0:
+        storage_list.append(MemoryProvider(f"cache/{dataset_id}"))
```

Comment:
This still doesn't mean anything, right? The path for MemoryProviders isn't getting used anywhere?

Reply:
Yup, the path is just to be consistent with other storage providers so that they can easily be used interchangeably. I've made the path an optional argument.